src/custom/state/orders/mocks.ts

```diff
@@ -0,0 +1,180 @@
+import { useEffect, useRef } from 'react'
+import { batch } from 'react-redux'
+
+import { Order, OrderStatus, OrderKind } from './actions'
+import { useActiveWeb3React } from 'hooks'
+import { useAddPendingOrder, usePendingOrders, useFulfillOrder } from './hooks'
+import { useSelectedTokenList } from 'state/lists/hooks'
+import { useAddPopup } from '@src/state/application/hooks'
+import { TokenInfo } from '@uniswap/token-lists'
+
+const randomNumberInRange = (min: number, max: number) => {
+  return Math.random() * (max - min) + min
+}
+
+// Infinity to not trigger condition
+const randomIntInRangeExcept = (min: number, max: number, exception = Infinity) => {
+  let num = Math.floor(randomNumberInRange(min, max))
+  // >= because we Math.floor
+  if (num >= exception) ++num
+  return num
+}
+
+const getRandomElementFromArray = <T>(array: T[]): T => {
+  const ind = randomIntInRangeExcept(0, array.length - 1)
+  return array[ind]
+}
+
+const getTwoRandomElementsFromArray = <T>(array: T[]): [T, T] => {
+  const ind1 = randomIntInRangeExcept(0, array.length - 1)
+  const ind2 = randomIntInRangeExcept(0, array.length - 1, ind1)
+
+  return [array[ind1], array[ind2]]
+}
+
+type MinTokenData = Pick<TokenInfo, 'symbol' | 'decimals' | 'address'>
+
+interface GenerateOrderParams extends Pick<Order, 'owner'> {
+  sellSymbol?: string
+  buySymbol?: string
+  buyToken: MinTokenData
+  sellToken: MinTokenData
+}
+
+// increment for OrderId
+let orderN = 1
+const generateOrderId = (ind: number) => {
+  return `OrderId_${ind}_`.padEnd(56 * 2, 'X')
+}
+
+const generateOrder = ({ owner, sellToken, buyToken }: GenerateOrderParams): Order => {
+  const sellAmount = (Math.random() * 1e18).toString(10) // in atoms
+  const buyAmount = (Math.random() * 1e18).toString(10) // in atoms
```

Comment:
why not using the utils?

Reply:
Don't care much about in the mock
`utils.parseEther` you mean? Not really necessary imho
It isn't even necessary to have these amounts in atoms in the mock, I just felt like adhering to the Order shape

And Order shape is changing already, like `0x` prefix...

Reply:
all good, lets not overdiscus mock mock